bin/morgue.js

```diff
@@ -7437,6 +7438,136 @@ async function metricsImporterCmd(argv, config) {
   await cli.routeMethod(argv);
 }
 
+function stabilityCreateMetric(coroner, argv, config) {
+  let universe = argv.universe;
+  if (!universe) {
+    universe = Object.keys(config.config.universes)[0];
+  }
+
+  let project = argv.project;
+
+  if (!universe) {
+    errx("--universe is required");
+  }
+
+  if (!project) {
+    errx("--project is required");
+  }
+
+  let bpg = coronerBpgSetup(coroner, argv);
+  let model = bpg.get();
+
+  /* Find universe. */
+  let uid = 0;
+  for (let u of model.universe) {
+    if (u.get("name") === universe) {
+      uid = u.get("id");
+    }
+  }
+
+  if (uid == 0) {
+    errx("Universe not found");
+  }
+
+  let pid = 0;
+  for (let p of model.project) {
+    if (p.get("universe") === uid && p.get("name") == project) {
+      pid = p.get("pid");
+    }
+  }
+
+  if (pid === 0) {
+    errx("Project not found");
+  }
+
+  const metricGroupName = argv["metric-group"];
+  if (!metricGroupName) {
+    errx("--metric-group is required");
+  }
+  if (Array.isArray(metricGroupName)) {
+    errx("SPecify only one --metric-group");
+  }
+
+  const name = argv.name;
+  if (!name) {
+    errx("--name is required");
+  }
+  if (Array.isArray(name)) {
+    errx("Only one --name allowed");
+  }
+
+  let metricGroupId = 0;
+
+  if (!model.metric_group) {
+    errx("No metric groups exist yet.");
+  }
+
+  for (let g of model.metric_group) {
+    if (g.get("name") === metricGroupName &&
+      g.get("project") === pid) {
+        metricGroupId = g.get("id");
+    }
+  }
+
+  if (metricGroupId === 0) {
+    errx("Metric group not found");
+  }
+
+  let attributes = argv.attribute;
+  if (attributes === undefined) {
+    /*
+     * 0 attributes are possible, though rare save for people integrating with
+     * the submission API directly.
+     */
+    attributes = [];
+  } else if (!Array.isArray(attributes)) {
+    attributes = [ attributes ];
+  }
+
+  let attributesObj = {};
+  for (let a of attributes) {
+    const parts = a.split(",");
+    if (parts.length != 2) {
```

Comment:
Should we validate if second part of `parts` array is empty? Example: `aaa,` will generate [`aaa`,``]

Reply:
I will for now, but it should be noted that empty strings are valid attributes, and it's not impossible that we'll go back on this later when someone asks for that.